tests/app/notifications/test_process_notification.py

```diff
@@ -0,0 +1,90 @@
+import pytest
+from boto3.exceptions import Boto3Error
+from sqlalchemy.exc import SQLAlchemyError
+
+from app.models import Template, Notification, NotificationHistory
+from app.notifications.process_notifications import (create_content_for_notification,
+                                                     persist_notification, send_notification_to_queue)
+from app.v2.errors import BadRequestError
+from tests.app.conftest import sample_notification, sample_template, sample_email_template
+
+
+def test_create_content_for_notification_passes(sample_email_template):
+    template = Template.query.get(sample_email_template.id)
+    content = create_content_for_notification(template, None)
+    assert content.replaced == template.content
+
+
+def test_create_content_for_notification_with_placeholders_passes(sample_template_with_placeholders):
+    template = Template.query.get(sample_template_with_placeholders.id)
+    content = create_content_for_notification(template, {'name': 'Bobby'})
+    assert content.content == template.content
+    assert 'Bobby' in content.replaced
+
+
+def test_create_content_for_notification_fails_with_missing_personalisation(sample_template_with_placeholders):
+    template = Template.query.get(sample_template_with_placeholders.id)
+    with pytest.raises(BadRequestError):
+        create_content_for_notification(template, None)
+
+
+def test_create_content_for_notification_fails_with_additional_personalisation(sample_template_with_placeholders):
+    template = Template.query.get(sample_template_with_placeholders.id)
```

Comment:
you don't need this query - just pass `sample_template_with_placeholders` in to the `create_content_for_notification` call